cpp/src/parquet/encoding.cc

```diff
@@ -2512,7 +2512,7 @@ struct ByteStreamSplitDecodePath<T, true> {
     // Stage 3: ABCD ABCD ABCD ABCD
     __m128i stage[num_streams_log2 + 1U][num_streams];
     const size_t half = num_streams / 2U;
-    for (size_t i = 0; i < num_blocks; ++i) {
+    for (int i = 0; i < num_blocks; ++i) {
```

Comment:
Why is an int better here? Add a comment?

Reply:
It's not better. Arrow often tends to use signed data types even when the value is always supposed to be non-negative. To avoid compiler errors on the remote server, I went with int always to avoid  static casting types everywhere which could be error prone on its own.